# **Waze**


# **Hypothesis testing**

**The purpose** of this project is to conduct a two-sample hypothesis test (t-test) to analyze the difference in the mean amount of rides between iPhone users and Android users.

*This Notebook has three parts:*

**Part 1:** Imports and data loading
* What data packages will be necessary for hypothesis testing?

**Part 2:** Conduct hypothesis testing
* How did computing descriptive statistics help analyze the data?

* How the null hypothesis and alternative hypothesis were formulated?

**Part 3:** Communicate insights with stakeholders

* What key business insight(s) emerged from the hypothesis test?

* What business recommendations can be proposed based on the results?





# **PACE stages**



## **PACE: Plan**

Consider the question to begin the analysis:
> Is there a statistically significant difference in mean amount of rides between iPhone® users and Android™ users?

### **Task 1. Imports and data loading**




In [1]:
import numpy as np
import pandas as pd
from scipy import stats

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
file_path = '/content/drive/MyDrive/Data Analytics/Main Projects/Waze project/waze_dataset.csv'
df = pd.read_csv(file_path)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14999 entries, 0 to 14998
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       14999 non-null  int64  
 1   label                    14299 non-null  object 
 2   sessions                 14999 non-null  int64  
 3   drives                   14999 non-null  int64  
 4   total_sessions           14999 non-null  float64
 5   n_days_after_onboarding  14999 non-null  int64  
 6   total_navigations_fav1   14999 non-null  int64  
 7   total_navigations_fav2   14999 non-null  int64  
 8   driven_km_drives         14999 non-null  float64
 9   duration_minutes_drives  14999 non-null  float64
 10  activity_days            14999 non-null  int64  
 11  driving_days             14999 non-null  int64  
 12  device                   14999 non-null  object 
dtypes: float64(3), int64(8), object(2)
memory usage: 1.5+ MB


In [5]:
df.head()

ID     label  sessions  drives  total_sessions  n_days_after_onboarding  \
0   0  retained       283     226      296.748273                     2276   
1   1  retained       133     107      326.896596                     1225   
2   2  retained       114      95      135.522926                     2651   
3   3  retained        49      40       67.589221                       15   
4   4  retained        84      68      168.247020                     1562   

   total_navigations_fav1  total_navigations_fav2  driven_km_drives  \
0                     208                       0       2628.845068   
1                      19                      64      13715.920550   
2                       0                       0       3059.148818   
3                     322                       7        913.591123   
4                     166                       5       3950.202008   

   duration_minutes_drives  activity_days  driving_days   device  
0              1985.775061             28            19  Android  
1              3160.472914             13            11   iPhone  
2              1610.735904             14             8  Android  
3               587.196542              7             3   iPhone  
4              1219.555924             27            18  Android

## **PACE: Analyze and Construct**




### **Task 2. Data exploration**


Binary-enconding the variable `device`:



In [6]:
map_dict = {"iPhone":1,"Android":2}
df["device_type"] = df["device"].replace(map_dict)
df.head()

ID     label  sessions  drives  total_sessions  n_days_after_onboarding  \
0   0  retained       283     226      296.748273                     2276   
1   1  retained       133     107      326.896596                     1225   
2   2  retained       114      95      135.522926                     2651   
3   3  retained        49      40       67.589221                       15   
4   4  retained        84      68      168.247020                     1562   

   total_navigations_fav1  total_navigations_fav2  driven_km_drives  \
0                     208                       0       2628.845068   
1                      19                      64      13715.920550   
2                       0                       0       3059.148818   
3                     322                       7        913.591123   
4                     166                       5       3950.202008   

   duration_minutes_drives  activity_days  driving_days   device  device_type  
0              1985.775061             28            19  Android            2  
1              3160.472914             13            11   iPhone            1  
2              1610.735904             14             8  Android            2  
3               587.196542              7             3   iPhone            1  
4              1219.555924             27            18  Android            2

The atention lies on the relationship between `device type` and the number of `drives`. One approach is to look at the average number of drives for each device type.

Calculating these averages:

In [7]:
df.groupby("device_type")["drives"].mean()

device_type
1    67.859078
2    66.231838
Name: drives, dtype: float64

Based on the averages shown, it appears that drivers who use an iPhone device to interact with the application have a higher number of drives on average. However, this difference might arise from random sampling, rather than being a true difference in the number of drives. To assess whether the difference is statistically significant, you can conduct a hypothesis test.


### **Task 3. Hypothesis testing**

The goal is to conduct a two-sample t-test. Recall the steps:

1.   State the null hypothesis and the alternative hypothesis
2.   Choose a signficance level
3.   Find the p-value
4.   Reject or fail to reject the null hypothesis

**Note:** This is a t-test for two independent samples. This is the appropriate test since the two groups are independent (Android users vs. iPhone users).

The hypotheses for this data project are:

* $H_0$ = The difference in the means is due tu chance. So, both means are equal.
* $H_A$ = The difference in the means is statistically significant.

Let's choose 5% as the significance level and proceed with a two-sample t-test.


**Technical note**: The default for the argument `equal_var` in `stats.ttest_ind()` is `True`, which assumes population variances are equal. This equal variance assumption might not hold in practice (that is, there is no strong reason to assume that the two groups have the same variance); you can relax this assumption by setting `equal_var` to `False`, and `stats.ttest_ind()` will perform the unequal variances $t$-test (known as Welch's `t`-test). Refer to the [scipy t-test documentation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_ind.html) for more information.


1. Isolate the `drives` column for iPhone users.
2. Isolate the `drives` column for Android users.
3. Perform the t-test


In [8]:
iphone_drives = df[df["device_type"] == 1]["drives"]

android_drives = df[df["device_type"] == 2]["drives"]

stats.ttest_ind(a = iphone_drives, b = android_drives, equal_var = False)

TtestResult(statistic=1.463523206885235, pvalue=0.143351972680206, df=11345.066049381952)

**Question:** Based on the p-value, do we reject or fail to reject the null hypothesis?

> The p-value of 0.14 is greater than the significance level of 0.05, so the tests fail to reject the null hypothesis. **This means that the difference between the means os drives by Android and Iphone users is due to chance.**


## **PACE: Execute**



### **Task 4. Communicate insights with stakeholders**

Consider the following question:


* What business insight(s) can be drawn from the result of the hypothesis test?

> *The key business insight is that drivers who use iPhone devices on average have a similar number of drives as those who use Androids.*

> *One potential next step is to explore what other factors influence the variation in the number of drives, and run additonal hypothesis tests to learn more about user behavior. Further, temporary changes in marketing or user interface for the Waze app may provide more data to investigate churn.*